File: solutions/0868. Binary Gap/0868.cpp
// Time Complexity: O(log⁡n)
// Space Complexity: O(1)
class Solution {
 public:
  int binaryGap(int n) {
    int ans = 0;

    for (int d = -32; n; n /= 2, ++d)
      if (n % 2 == 1) {
        ans = max(ans, d);
        d = 0;
      }

    return ans;
  }
};

File: solutions/0868. Binary Gap/0868.rs
// Time Complexity: O(log⁡n)
// Space Complexity: O(1)
impl Solution {
    pub fn binary_gap(n: i32) -> i32 {
        let mut ans = 0;
        let mut d = -32;

        let mut num = n;
        while num != 0 {
            if num % 2 == 1 {
                ans = ans.max(d);
                d = 0;
            }
            num /= 2;
            d += 1;
        }

        ans
    }
}
